packages/client/src/connection.ts

```diff
@@ -100,6 +100,14 @@ export interface ConnectionOptions {
    */
   metadata?: Metadata;
 
+  /**
+   * API key for Temporal. This becomes the "Authorization" HTTP header with "Bearer " prepended.
+   * This is only set if RPC metadata doesn't already have an "authorization" key.
+   *
+   * In order to dynamically set metadata, use {@link Connection.withApiKey} or {@link Connection.setApiKey}
+   */
+  apiKey?: string;
```

Comment:
If the setter accepts a function, you should accept a function here too:

```suggestion
  apiKey?: string | (() => string);
```

Reply:
Sure, I considered it but thought they'd probably just set it after construction if they need it dynamic, but, why not.